examples/custom-ipfs-repo/index.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const series = require('async/series')
+const IPFS = require('ipfs')
+const Repo = require('ipfs-repo')
+
+// Create our custom options
+const customRepositoryOptions = {
+
+  /**
+   * Storage Backends are fully customizable. Each backend can be stored in seperate services,
+   * or in a single service. Options can be passed into the datastores via the storageBackendOptions
+   * property shown below.
+   */
+  storageBackends: {
+    root: require('datastore-fs'), // version and config data will be saved here
+    blocks: require('datastore-fs'),
+    keys: require('datastore-fs'),
+    datastore: require('datastore-fs')
+  },
+
+  /**
+   * Storage Backend Options will get passed into the instantiation of their counterpart
+   * in Storage Backends. If you create a custom datastore, this is where you can pass in
+   * custom insantiation. You can see an S3 datastore example at:
+   * https://github.com/ipfs/js-datastore-s3/tree/master/examples/full-s3-repo
+   */
+  storageBackendOptions: {
+    blocks: {
+      sharding: true,
+      extension: '.data'
+    }
+  },
+
+  // false will disable locking, you can also pass in a custom locker
+  locker: false
+}
+
+// Initialize our repo and IPFS node
+const myRepo = new Repo('/tmp/custom-repo/.ipfs', customRepositoryOptions)
+const node = new IPFS({
+  repo: myRepo
```

Comment:
@jacobheun I think it makes sense to remove the step of requiring 'ipfs-repo' and set it up manually by just adding a `repoOptions` object to the IPFS constructor options that lets us pass the storagebackends directly.